pymc_experimental/distributions/multivariate/r2d2m2cp.py

```diff
@@ -59,16 +67,141 @@ def _R2D2M2CP_beta(
     """
     tau2 = r2 / (1 - r2)
     explained_variance = phi * pt.expand_dims(tau2 * output_sigma**2, -1)
-    mu_param, std_param = _psivar2musigma(psi, explained_variance)
+    mu_param, std_param = _psivar2musigma(psi, explained_variance, psi_mask=psi_mask)
     if not centered:
         with pm.Model(name):
-            raw = pm.Normal("raw", dims=dims)
+            if psi_mask is not None and psi_mask.any():
+                # limit case where some probs are not 1 or 0
+                # setsubtensor is required
+                r_idx = psi_mask.nonzero()
+                with pm.Model("raw"):
+                    raw = pm.Normal("masked", shape=len(r_idx[0]))
+                    raw = pt.set_subtensor(pt.zeros_like(mu_param)[r_idx], raw)
+                raw = pm.Deterministic("raw", raw, dims=dims)
+            elif psi_mask is not None:
+                # all variables are deterministic
+                raw = pt.zeros_like(mu_param)
+            else:
+                raw = pm.Normal("raw", dims=dims)
         beta = pm.Deterministic(name, (raw * std_param + mu_param) / input_sigma, dims=dims)
     else:
-        beta = pm.Normal(name, mu_param / input_sigma, std_param / input_sigma, dims=dims)
+        if psi_mask is not None and psi_mask.any():
+            # limit case where some probs are not 1 or 0
+            # setsubtensor is required
+            r_idx = psi_mask.nonzero()
+            with pm.Model(name):
+                mean = (mu_param / input_sigma)[r_idx]
+                sigma = (std_param / input_sigma)[r_idx]
+                masked = pm.Normal(
+                    "masked",
+                    mean,
+                    sigma,
+                    shape=len(r_idx[0]),
+                )
+                beta = pt.set_subtensor(mean, masked)
+            beta = pm.Deterministic(name, beta, dims=dims)
+        elif psi_mask is not None:
+            # all variables are deterministic
+            beta = pm.Deterministic(name, (mu_param / input_sigma), dims=dims)
+        else:
+            beta = pm.Normal(name, mu_param / input_sigma, std_param / input_sigma, dims=dims)
     return beta
 
 
+def _broadcast_as_dims(*values, dims):
+    model = pm.modelcontext(None)
+    shape = [len(model.coords[d]) for d in dims]
+    ret = tuple(np.broadcast_to(v, shape) for v in values)
+    # strip output
+    if len(values) == 1:
+        ret = ret[0]
+    return ret
```

Comment:
What if you have mutable dims?

Reply:
this dims can't be mutable since they are used to create RVs, no?

Reply:
Yes they can. RVs can have dynamic shapes (just not number of dimensions)

Reply:
these particular ones are supposed to be static shape parameters anyway..., like parameters in a linear regression.

Reply:
What would you suggest to do?

Reply:
@ricardoV94 it seems to be impossible to implement in a clean way with mutable dims, but I bet in future it can be refactored into some more generic implementation. The main challenge is to take in account the limit case where random variable is partial

Reply:
I doubt this particular use-
case is frequent, we can just create a new issue for that

Reply:
@ricardoV94 this is ready for review, I've added checks for that and opened an issue